### DISCUSSION TITLE
How to share the workspaces between different deployment environments.

### DISCUSSION THREAD

**Discussion by recursively** [illacloud/illa-builder#1126] [General]
I'm just curious are there any ways to share the workspaces between different deployment environments? I just created a temporary workspace and performed the demo testing on my PC,  now I want to deploy the workspace on my remote server, do I need to build everything from scratch? It's killing me, to be honest.

**Comment by wenzhuo2022** [comment]
Hello, I am sorry for the bad experience brought you. At present, this is really the only way to do it. In future versions, we will support data migration and export configuration.

**Reply by recursively** [reply]
Alright, what if my local illa-builder image was destroyed but the data was kept safe because the locally mounted volume wasn't affected? Is there any way to restore the workspace I built previously?

**Reply by karminski** [reply]
@recursively Hi, you can re-pull the illa-builder image and start it with your local mount address. It will read the local volume data normally. **(Please back up your data before this, just in case)** 

For migration, you can try to move the database file to your server and mount it into the container for now, it's tricky but works. We will support data migration and export configuration later.
